src/Microsoft.OpenApi/Validations/Rules/JsonSchemaRules.cs

```diff
@@ -34,6 +34,22 @@ public static class JsonSchemaRules
 
                     context.Exit();
 
+                    // examples
+                    context.Enter("examples");
+
+                    if (jsonSchema.GetExamples() != null)
+                    {
+                        for (int i = 0; i < jsonSchema.GetExamples().Count(); i++)
```

Comment:
```suggestion
                        var examplesCount = examples.Count();
                        for (int i = 0; i < examplesCount; i++)
```

(this avoids enumerating the examples at each loop evaluation)